api/client/credentials.go

```diff
@@ -313,28 +291,6 @@ type profileCreds struct {
 	profile *profile.Profile
 }
 
-// Dialer is used to dial a connection to an Auth server.
-func (c *profileCreds) Dialer(cfg Config) (ContextDialer, error) {
-	sshConfig, err := c.SSHClientConfig()
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	tlsConfig, err := c.profile.TLSConfig()
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	return NewProxyDialer(
-		*sshConfig,
-		cfg.KeepAlivePeriod,
-		cfg.DialTimeout,
-		c.profile.WebProxyAddr,
-		cfg.InsecureAddressDiscovery,
-		WithTLSConfig(tlsConfig),
```

Comment:
On this topic, the WithTLSConfig was added in https://github.com/gravitational/teleport/pull/22439 to support HTTPS_PROXY, and this TLS config shouldn't be used anywhere else. 

cc @atburke 

My understanding is there is no need to pass a full TLS config for HTTPS_PROXY. It doesn't make sense to me to pass custom cert or cas when dialing the proxyURL, and toggling insecure is probably enough here.